tests/cnf/core/network/metallb/tests/frrk8-tests.go

```diff
@@ -0,0 +1,771 @@
+					externalAdvertisedIPv6Routes, true)
+
+				By("Creating BGP Peers")
+				createBGPPeerAndVerifyIfItsReady("172.16.0.1", "", tsparams.RemoteBGPASN,
+					true, frrk8sPods)
+
+				By("Checking that BGP session is established and up")
+				verifyMetalLbBGPSessionsAreUPOnFrrPod(frrPod, removePrefixFromIPList(ipv4NodeAddrList))
+
+				By("Validating BGP route prefix")
+				validatePrefix(frrPod, "IPv4", removePrefixFromIPList(nodeAddrList), addressPool, 32)
+
+				By("Create a frrconfiguration allow all for EBGP multihop")
+				err = createFrrConfiguration("frrconfig-allow-all", "172.16.0.1", 64501, nil, true)
+				Expect(err).ToNot(HaveOccurred(), "Fail to create an allow-all FRRconfiguration")
+
+				By("Verify that the node FRR pods advertises two routes")
+				err = verifyExternalAdvertisedRoutes(frrPod, ipv4NodeAddrList, externalAdvertisedIPv4Routes)
+				Expect(err).ToNot(HaveOccurred(), "Fail to find advertised routes")
+
+				By("Validate BGP received routes")
+				Eventually(func() string {
+					// Get the routes
+					routes, err := frr.VerifyBGPReceivedRoutesOnFrrNodes(frrk8sPods)
+					Expect(err).ToNot(HaveOccurred(), "Failed to verify BGP routes")
+
+					// Return the routes to be checked
+					return routes
+				}, 60*time.Second, 5*time.Second).Should(SatisfyAll(
+					ContainSubstring(externalAdvertisedIPv4Routes[0]), // First IP address
+					ContainSubstring(externalAdvertisedIPv4Routes[1]), // Second IP address
+				), "Fail to find all expected received routes")
+			})
+	})
+
+})
+
+func createNewMetalLbDaemonSetAndWaitUntilItsRunningWithAlwaysBlock(timeout time.Duration,
+	nodeLabel map[string]string, prefixes []string) error {
+	By("Verifying if metalLb daemonset is running")
+
+	metalLbIo, err := metallb.Pull(APIClient, tsparams.MetalLbIo, NetConfig.MlbOperatorNamespace)
+
+	if err == nil {
+		glog.V(90).Infof("MetalLb daemonset is running. Removing daemonset.")
+
+		_, err = metalLbIo.Delete()
+
+		if err != nil {
+			return err
+		}
+	}
+
+	glog.V(90).Infof("Create new metalLb speaker's daemonSet.")
+
+	metalLbIo = metallb.NewBuilder(
+		APIClient, tsparams.MetalLbIo, NetConfig.MlbOperatorNamespace, nodeLabel)
+
+	metalLbIo.WithFRRConfigAlwaysBlock(prefixes)
+	_, err = metalLbIo.Create()
+
+	if err != nil {
+		return err
+	}
+
+	var metalLbDs *daemonset.Builder
+
+	err = wait.PollUntilContextTimeout(
+		context.TODO(), 3*time.Second, timeout, true, func(ctx context.Context) (bool, error) {
+			metalLbDs, err = daemonset.Pull(APIClient, tsparams.MetalLbDsName, NetConfig.MlbOperatorNamespace)
+			if err != nil {
+				glog.V(90).Infof("Error to pull daemonset %s namespace %s, retry",
+					tsparams.MetalLbDsName, NetConfig.MlbOperatorNamespace)
+
+				return false, nil
+			}
+
+			return true, nil
+		})
+
+	if err != nil {
+		return err
+	}
+
+	glog.V(90).Infof("Waiting until the new metalLb daemonset is in Ready state.")
+
+	if metalLbDs.IsReady(timeout) {
+		return nil
+	}
+
```

Comment:
wdyt about also verifying that the frr-k8s daemonset is ready?

Reply:
Added